source/renderer/app/components/wallet/summary/WalletSummaryAsset.js

```diff
@@ -0,0 +1,213 @@
+// @flow
+import React, { Component } from 'react';
+import { observer } from 'mobx-react';
+import { defineMessages, intlShape } from 'react-intl';
+import { Button } from 'react-polymorph/lib/components/Button';
+import { PopOver } from 'react-polymorph/lib/components/PopOver';
+import classNames from 'classnames';
+import { get } from 'lodash';
+import SVGInline from 'react-svg-inline';
+import styles from './WalletSummaryAsset.scss';
+import Asset from '../../assets/Asset';
+import AssetAmount from '../../assets/AssetAmount';
+import AssetContent from '../../assets/AssetContent';
+import type { AssetToken } from '../../../api/assets/types';
+import arrow from '../../../assets/images/collapse-arrow-small.inline.svg';
+import warningIcon from '../../../assets/images/asset-token-warning-ic.inline.svg';
+
+const messages = defineMessages({
+  tokenSendButton: {
+    id: 'wallet.summary.asset.tokenSendButton',
+    defaultMessage: '!!!Send',
+    description: 'Send button on Wallet summary assets page',
+  },
+  amountLabel: {
+    id: 'wallet.summary.asset.amountLabel',
+    defaultMessage: '!!!Amount',
+    description: 'Amount label on Wallet summary assets page',
+  },
+  settingsButtonLabel: {
+    id: 'wallet.summary.asset.settings.button.label',
+    defaultMessage: '!!!Settings',
+    description: 'Settings label on Wallet summary assets page',
+  },
+  settingsWarningPopOverAvailable: {
+    id: 'assets.warning.available',
+    defaultMessage:
+      '!!!Recommended configuration for decimal places for this native token is available.',
+    description: 'Asset settings recommended pop over content',
+  },
+  settingsWarningPopOverNotUsing: {
+    id: 'assets.warning.notUsing',
+    defaultMessage:
+      '!!!You are not using the recommended decimal place configuration for this native token.',
+    description: 'Asset settings recommended pop over content',
+  },
+});
+
+type Props = {
+  asset: AssetToken,
+  onOpenAssetSend: Function,
+  onCopyAssetItem: Function,
+  onAssetSettings: Function,
+  anyAssetWasHovered: boolean,
+  isLoading: boolean,
+  assetSettingsDialogWasOpened: boolean,
+};
+
+type State = {
+  isExpanded: boolean,
+};
+
+@observer
+export default class WalletSummaryAsset extends Component<Props, State> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  state = {
+    isExpanded: false,
+  };
+
+  toggleIsExpanded = () => {
+    this.setState((prevState) => ({
+      isExpanded: !prevState.isExpanded,
+    }));
+  };
+
+  renderHeader = () => {
```

Comment:
if we are splitting up parts of the component into separate functions because it gets too large otherwise - that's a good indication for creating more, smaller components that can be composed. E.g `WalletSummaryAssetHeader` etc.

Reply:
I definitely agree we need to fragment more our components, as there are some monsters there, but not sure I agree it's the case, here. It's a very concise and interconnected component and its parts won't be used anywhere else. Don't see how it'd be beneficial to break it up into multiple files 🤔

Reply:
I've tried splitting up, but it honestly looked so messy, as the components were not meant to be used outside WalletSummaryAsset.js. I would either keep them inside or create a different naming/folder structure, so it would be clear they are internal components

Reply:
Ok, yeah we need to find a nicer way to write these big components … i agree that splitting them up not necessarily solves the issue - especially if these components are only used once for this case